Src/AutoFixture.xUnit.net.UnitTest/FrozenAttributeTest.cs

```diff
@@ -82,18 +69,129 @@ public void GetCustomizationWithSpecificRegisteredTypeReturnsCorrectResult()
+            // Verify outcome
+            var freezer = Assert.IsAssignableFrom<FreezeOnMatchCustomization>(customization);
+            var matcher = Assert.IsType<OrRequestSpecification>(freezer.Matcher);
+            Assert.NotEmpty(matcher.Specifications.OfType<DirectBaseTypeSpecification>());
+            // Teardown
+        }
+
+        [Fact]
+        public void GetCustomizationWithMatchingByImplementedInterfacesShouldMatchByImplementedInterfaces()
+        {
+            // Fixture setup
+            var sut = new FrozenAttribute(Matching.ImplementedInterfaces);
+            // Exercise system
+            var customization = sut.GetCustomization(AParameter<object>());
+            // Verify outcome
+            var freezer = Assert.IsAssignableFrom<FreezeOnMatchCustomization>(customization);
+            var matcher = Assert.IsType<OrRequestSpecification>(freezer.Matcher);
+            Assert.NotEmpty(matcher.Specifications.OfType<ImplementedInterfaceSpecification>());
+            // Teardown
+        }
+
+        [Fact]
+        public void GetCustomizationWithMatchingByParameterNameShouldMatchByParameter()
+        {
+            // Fixture setup
+            var sut = new FrozenAttribute(Matching.ParameterName, "parameter");
+            // Exercise system
+            var customization = sut.GetCustomization(AParameter<object>());
+            // Verify outcome
+            var freezer = Assert.IsAssignableFrom<FreezeOnMatchCustomization>(customization);
+            var matcher = Assert.IsType<OrRequestSpecification>(freezer.Matcher);
+            Assert.NotEmpty(matcher.Specifications.OfType<ParameterSpecification>());
+            // Teardown
+        }
+
+        [Fact]
+        public void GetCustomizationWithMatchingByPropertyNameShouldMatchByProperty()
+        {
+            // Fixture setup
+            var sut = new FrozenAttribute(Matching.PropertyName, "Property");
+            // Exercise system
+            var customization = sut.GetCustomization(AParameter<object>());
+            // Verify outcome
+            var freezer = Assert.IsAssignableFrom<FreezeOnMatchCustomization>(customization);
+            var matcher = Assert.IsType<OrRequestSpecification>(freezer.Matcher);
+            Assert.NotEmpty(matcher.Specifications.OfType<PropertySpecification>());
+            // Teardown
+        }
+
+        [Fact]
+        public void GetCustomizationWithMatchingByFieldNameShouldMatchByField()
+        {
+            // Fixture setup
+            var sut = new FrozenAttribute(Matching.FieldName, "Field");
+            // Exercise system
+            var customization = sut.GetCustomization(AParameter<object>());
+            // Verify outcome
+            var freezer = Assert.IsAssignableFrom<FreezeOnMatchCustomization>(customization);
+            var matcher = Assert.IsType<OrRequestSpecification>(freezer.Matcher);
+            Assert.NotEmpty(matcher.Specifications.OfType<FieldSpecification>());

```

Comment:
The `TargetType` and `TargetName` properties in `FieldSpecification` were deprecated in 1925183 so we only look at the type.